cmake-scripts/version.cmake

```diff
@@ -0,0 +1,298 @@
+#
+# version.cmake has the job of setting five variables:
+# - SHA1_HASH, VERSION_MAJOR, VERSION_MINOR, VERSION_PATCH, AND VERSION_LABEL
+#
+# It calculates these from the information in LatestRelease.cmake, and by running
+# a several git commands.
+#
+# Although it is used in CMakeLists.txt, version.cmake is also run from github actions
+# as a standalone cmake script, so that our actions have access to the version information as well.
+#
+
+#
+# OVERVIEW
+#
+# If our git HEAD commit matches LATEST_RELEASE_TAG,
+# then our version is exactly LATEST_RELEASE_TAG and we are done.
+#
+# Otherwise we're not exactly on a release version, and our
+# version is going to take the form:
+#
+#   X.Y.Z-C-HHHHHHH
+#
+# Where:
+# - X is the major version number from the most recent release
+# - Y is the minor version number from the most recent release
+# - Z is *one more than* the patch version number from the most recent release.
+# - C is the number of commits since the release was made, counted as described below.
+# - and HHHHHHHH is an abbreviated git commit hash.
+#
+#
+# A NOTE ON VERSION LABELS.
+# as a technical note, -C-HHHHHHHH together form the "version label".
+# any version label (e.g. "-rc1") from the latest release's tag is also
+# carried forward, and so a version can look like 0.8.1-rc1-29-01dcec0a.
+#
+#
+# COUNTING COMMITS.
+# We use git to count the commits. There is one twist, because of versions like
+# 0.8.0 (where commits were applied to the release, but not master--)
```

Comment:
So, to be clear, we can and should merge the 0.8.0 branch back into master, which should solve this problem? Release branches should always be a subset of master, I think?

Reply:
Yes, release branch is supposed to be merged to master.

Reply:
After 0.8.0 is merged into master, I'll simplify this PR by removing LATEST_RELEASE_FORKED_FROM and (if my next idea on how to abuse git fetches works out) removing LatestRelease.cmake